libeos-updater-flatpak-installer/tests/installer.c

```diff
@@ -44,28 +185,59 @@ flatpak_deployments_fixture_setup (FlatpakDeploymentsFixture  *fixture,
   g_autofree gchar *flatpak_deployments_path = g_dir_make_tmp ("eos-updater-test-flatpak-deployments-XXXXXX", &error);
   g_autoptr(GFile) flatpak_deployments_directory = g_file_new_for_path (flatpak_deployments_path);
   g_autoptr(GFile) flatpak_build_dir = g_file_get_child (flatpak_deployments_directory, "flatpak");
-  const gchar *flatpak_names[] = {
-    "org.test.Test",
-    "org.test.Test2",
-    "org.test.Test3",
-    "org.test.Preinstalled",
-    NULL
+  g_autofree gchar *flatpak_installer_tmpdir = g_dir_make_tmp ("eos-updater-test-flatpak-installer-tmpdir-XXXXXX", &error);
+  g_autofree gchar *source_gpg_home_path = g_test_build_filename (G_TEST_DIST, "..", "..", "tests", "gpghome", NULL);
+  g_autoptr(GFile) gpg_key_file = NULL;
+  g_autofree gchar *keyid = NULL;
+  const FlatpakInfo flatpak_infos[] = {
+    { "org.test.Runtime", FLATPAK_INSTALL_INFO_TYPE_RUNTIME },
+    { "org.test.Test", FLATPAK_INSTALL_INFO_TYPE_APP },
+    { "org.test.Test2", FLATPAK_INSTALL_INFO_TYPE_APP },
+    { "org.test.Test3", FLATPAK_INSTALL_INFO_TYPE_APP },
+    { "org.test.Preinstalled", FLATPAK_INSTALL_INFO_TYPE_APP },
+    { "org.test.PreinstalledTestHavingExtension", FLATPAK_INSTALL_INFO_TYPE_APP },
+    { "org.test.PreinstalledTestHavingExtension.Extension", FLATPAK_INSTALL_INFO_TYPE_EXTENSION },
+    { NULL, 0 }
   };
   const gchar *preinstall_flatpak_names[] = {
+    "org.test.Runtime",
     "org.test.Preinstalled",
+    "org.test.PreinstalledTestHavingExtension",
+    "org.test.PreinstalledTestHavingExtension.Extension",
     NULL
   };
+  const ExtensionInfo extensions[] = {
+    {
+      "org.test.PreinstalledTestHavingExtension",
+      "org.test.PreinstalledTestHavingExtension.Extension",
+      FLATPAK_EXTENSION_POINT_AUTODELETE
+    },
+    { NULL, NULL }
+  };
```

Comment:
I know, this is getting dangerously close to the existing `FlatpakInstallInfo` and friends structs. I wanted to have these ones since they are easier to populate declaratively.